main.cpp

```diff
@@ -1,23 +1,36 @@
-#include "queue.h"
 #include <iostream>
+#include "queue.h"
 using namespace std;
-int main()
-{
-	int c, b;
-	data a[12], d;
-	ArrayQueue q(20);
-	cout << "press 1 if you want ArrayQueue, press 2 if you want ListQueue";
-	cin >> b;
-	if (b == 2)
-		ListQueue q;
-	for (c = 0;c < 10;c++)
-	{
-		cin >> d;
-		q.enqueue(d);
-	
-	}
-	for (c = 0;c < 10;c++)
-	{
-		cout << q.denqueue() << " ";
-	}
+
+ int main()
+ {
+    char c;
+    std::cout << "Type of queue (a - array, l - list): ";
+    std::cin >> c;
+	ArrayQueue queue(20);
+    if (c == 'a')
+	ArrayQueue queue(20);
+    else if (c == 'l')
+        ListQueue queue;
+    else 
+    {
+        std::cout << "Bad type, exit" << std::endl;
+        return 1;
+    }
+
+    for (c = 0;c < 10;c++)
+    {
+	cin >> d;
+	queue.enqueue(d);
+    }
+    for (c = 0;c < 10;c++)
+    {
+	cout << queue.denqueue() << " " 
+	cout << queue.getLength << endl;
+    }
+    delete &queue;

```

Comment:
You shouldn't delete objects that weren't created by `new` operator. It could lead to really devastating consequences.
Objects on the stack would be deleted at the end of its scope (and destructor will be called). 

Currently, here we have memory leak. Please, solve this problem with proper destructor.